master/static/srv/shell-entrypoint.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+
+set -e
+
+export PATH="/run/determined/pythonuserbase/bin:$PATH"
+
+# Unlike trial and notebook entrypoints, the HOME directory does not need to be
+# modified in this entrypoint because the HOME in the user's ssh session is set
+# by sshd at a later time.
+
+python3.6 -m pip install --user /opt/determined/wheels/determined*.whl
+
+# Prepend each key in authorized_keys with a set of environment="KEY=VALUE"
+# options to inject environment variables into the eventual ssh session via an
+# option in the authorized keys file.  See syntax described in `man 8 sshd`
+#
+# After openssh 8+ is the only version of openssh supported (that is, after we
+# only support ubuntu >= 20.04), we can use the more obvious SetEnv option and
+# skip this awkwardness.
+vars='
+    PATH
+    PYTHONUSERBASE
+    DET_TASK_ID
+    DET_CLUSTER_ID
+    DET_MASTER_ID
+    DET_MASTER
+    DET_MASTER_HOST
+    DET_MASTER_ADDR
+    DET_MASTER_PORT
+    DET_AGENT_ID
+    DET_CONTAINER_ID
+    DET_SLOT_IDS
+    DET_USE_GPU
+    CONDA_DIR
+    PYTHONUNBUFFERED
+    PYTHONFAULTHANDLER
+    PYTHONHASHSEED
+'
+options="$(
+    for var in $vars ; do
+        # Note that the syntax ${!var} is for a double dereference.
+        echo -n "environment=\"$var=${!var}\","
+    done
+)"
+sed -E -i -e "s|(.*)|$options \1|" "/run/determined/ssh/authorized_keys"
```

Comment:
question: I'm not that familiar with `sed`. Can you explain this command?

Reply:
Oh, I didn't think of matching to the beginning of the line.  That would have been a lot neater.